.clang-tidy

```diff
@@ -0,0 +1,25 @@
+Checks:            clang-*,
+                   -clang-analyzer-optin.portability.UnixAPI,
+                   -clang-analyzer-unix.Malloc,
+                   -clang-diagnostic-pragma-once-outside-header,
+                   cppcoreguidelines-pro-type-member-init,
+                   cppcoreguidelines-pro-type-static-cast-downcast,
+                   misc-*,
+                   -misc-non-private-member-variables-in-classes,
+                   modernize-*,
+                   -modernize-avoid-c-arrays,
+                   -modernize-return-braced-init-list,
+                   -modernize-use-trailing-return-type,
+                   llvm-include-order,
+                   performance-*,
+                   -performance-no-int-to-ptr,
+                   portability-*,
+                   readability-*,
+                   -readability-convert-member-functions-to-static,
+                   -readability-function-cognitive-complexity,
+                   -readability-inconsistent-declaration-parameter-name,
+                   -readability-magic-numbers,
+                   -readability-make-member-function-const,
+                   -readability-simplify-boolean-expr,
```

Comment:
Note: This list currently contains a lot of exclusion, because fixes for them require non-trivial changes in the code, and I didn't want to include them in this mass PR (in which 95% of changes is automatically applied suggestions from `clang-tidy`).